Plugins/Tweaks/Tweaks/HideClassesOnCharList.cpp

```diff
@@ -0,0 +1,64 @@
+#include "Tweaks/HideClassesOnCharList.hpp"
+
+#include "Services/Hooks/Hooks.hpp"
+
+#include "API/CAppManager.hpp"
+#include "API/CNWSCreature.hpp"
+#include "API/CNWSMessage.hpp"
+#include "API/CServerExoApp.hpp"
+#include "API/Functions.hpp"
+#include "API/Globals.hpp"
+
+namespace Tweaks {
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+
+HideClassesOnCharList::HideClassesOnCharList(ViewPtr<Services::HooksProxy> hooker)
+{
+    hooker->RequestExclusiveHook<API::Functions::CNWSMessage__SendServerToPlayerPlayModuleCharacterListResponse>
+        (&SendServerToPlayerPlayModuleCharacterListResponseHook);
+}
+
+int32_t HideClassesOnCharList::SendServerToPlayerPlayModuleCharacterListResponseHook(
+    CNWSMessage* thisPtr, Types::PlayerID playerId, Types::ObjectID charId, int32_t add)
+{
+    thisPtr->CreateWriteMessage(sizeof(playerId), playerId, true);
+
+    thisPtr->WriteBOOL(add);
+    thisPtr->WriteDWORD(charId, 32);
+
+    if (add)
+    {
+        CNWSCreature* creature = Globals::AppManager()->m_pServerExoApp->GetCreatureByGameObjectID(charId);
+        if (creature)
+        {
+            thisPtr->WriteCExoLocStringServer(creature->GetFirstName(), creature->GetGender());
+            thisPtr->WriteCExoLocStringServer(creature->GetLastName(), creature->GetGender());
+
+            uint16_t portraitId = creature->GetPortraitId();
+            thisPtr->WriteWORD(portraitId, 16);
+            if (portraitId >= 0xFFFE)
+            {
+                thisPtr->WriteCResRef(creature->GetPortrait(), 16);
+            }
+
+            thisPtr->WriteBYTE(0, 8);
+        }
+        else
+        {
+            return 0;
+        }
+    }
+
+    unsigned char* message;
```

Comment:
Nittiest of the nits, but to me `char` suggests a human readable string. The message is just a byte buffer, so `uint8_t` is a better fit semantically.

Reply:
I agree, `uint8_t` does fit better. Or even `std::byte` (if we had C++17). I just copied the API here, but I haven't done the same for `Types::PlayerID playerId` and `Types::ObjectID charId` so I'll change it.